### PR TITLE
Print all loaded engines with openssl list -engines command

### PR DESCRIPTION
**Pull request by t8m**
This adds functionality that was patched into the Fedora and RHEL openssl for a long time.
##### Checklist

* [x]  documentation is added or updated

### CODE REVIEW

**Comment by levitte** [comment]
I think it would be much more appropriate to add another option to `openssl list`, say `-list-engines`?

**Comment by levitte** [comment]
... or just `-engines`

**Comment by t8m** [comment]
OK, makes sense, I'll adjust the PR.

**Comment by t8m** [comment]
@levitte Is it acceptable this way?

File: apps/openssl.c [hunk 1]
```diff
@@ -837,6 +843,22 @@ static int SortFnByName(const void *_f1, const void *_f2)
...
+    ENGINE *e;
+
+    BIO_puts(bio_out, "Engines:\n");
+    e = ENGINE_get_first();
+    while (e) {
+        BIO_printf(bio_out, "%s\n", ENGINE_get_id(e));
+        e = ENGINE_get_next(e);
+    }
+#else
+    BIO_puts(bio_out, "Engines support is disabled.\n");
```
Comment (mspncp) [inline apps/openssl.c]: „Engine support ...“ ?

Comment (t8m) [reply apps/openssl.c]: OK, that looks better.

**Comment by mspncp** [comment]
Merged to ebb7823e14596ad07fdc7d2ed0a267815f545927, thanks!

**Comment by romen** [comment]
What are your feelings about backporting this to 1.1.1?

**Comment by mspncp** [comment]
Personally, I think it is a useful feature and the chance to break existing code is practically zero. But after all, it’s a new feature, so I would need affirmation from an @openssl/omc member that a backport doesn‘t violate our policies.

**Comment by levitte** [comment]
For patch updates of a release, this is what the [release strategy](https://www.openssl.org/policies/releasestrat.html) says:

> * PATCH: Bug fix releases will increment this number. We also allow backporting of accessor functions in these releases.

I doubt any of that makes it justifiable to backport this additional feature.

**Comment by mspncp** [comment]
That’s about what I expected. Thanks for your quick response.

**Pull request closed without merging**